### DISCUSSION TITLE
zh/kernel/onboarding/slides/ckb-dynamic-difficulty-adjustment/

### DISCUSSION THREAD

**Discussion by eval-exec** [ckb-learn/ckb-learn.github.io#13] [General]
- [CKB Dynamic Difficulty Adjustment.pptx](https://github.com/ckb-learn/ckb-learn.github.io/files/9617630/CKB.Dynamic.Difficulty.Adjustment.pptx)
- [CKB Two-Step Transaction Confirmation.pptx](https://github.com/ckb-learn/ckb-learn.github.io/files/9617631/CKB.Two-Step.Transaction.Confirmation.pptx)

// 2022.9.20 难度调整

**Comment by eval-exec** [comment]
问：以太坊的叔块和 ckb 的叔块有什么共同点和区别？

共同点：
1.叔块的存在能为主链的安全做贡献，引用叔块越多，主链就越重，越安全。
2.可以对抗大型中心化矿池的势力，给那些挖出叔块的矿工也给奖励，照顾一下身家小的矿工，支持去中心化。

区别：
1. 叔块里的 Tx：
ckb 的叔块里的 Tx ProposalId 是有用的，可以被后面的矿工打包进去的； ckb 的叔块能帮着把算力统计进来，然后用作下一个 epoch 的难度调整。
而 ETH 的叔块里的 Tx 是无效的。

2. 叔块奖励的发放不同： 
以太坊上，挖出叔块的矿工奖励和叔块被引用进来的高度有关，高度差越小，挖出叔块的矿工得到的奖励越多；把叔块引用进主链的矿工也有奖励，奖励和引用进来的叔块数量有关。
而 ckb 上，如果叔块里的 proposalId 被后续的块使用了，那么挖出叔块的矿工有可能会分到提交 proposalId 的那部分的奖励。

请问哪里有理解不对的或不完整地方吗？

**Reply by doitian** [reply]
> 挖出叔块的矿工有可能会分到提交 proposalId 的那部分的奖励。

叔块的矿工没有奖励，奖励是给打包叔块的那个块的矿工

**Reply by zhangsoledad** [reply]
叔块是没有奖励的，这也是本质不同导致的。
以太坊的叔块 有介绍过了 就是 GHOST 协议的定义，
而 CKB的叔块，是为了 NC-MAX需要观察叔块率而设置，即叔块本身没有对网络作贡献，但是打包叔块的块有贡献，所以打包叔块也有间接的奖励，叔块的 proposal zone 会被并入。

**Reply by zhangsoledad** [reply]
proposal 和 交易手续费挂钩。

**Comment by felicityin** [comment]
Why there is no output of cellbase in blocks 1 to 11 on ckb?

区块 N 的 Cellbase Output 是发放给 N - 11 区块的生产者的
